scrub_hub_project/scrub_hub_chat/views.py

```diff
@@ -1,28 +1,35 @@
-from django.http import JsonResponse, HttpResponseBadRequest, HttpResponseForbidden
-from django.shortcuts import render
 import datetime
+import base64
+import json
 
 from .models import Conversation, ConversationParticipant, Message
 from authenticate.models import CustomUser as User
+
 from django.contrib.auth.decorators import login_required
+from django.http import JsonResponse, HttpResponseBadRequest, HttpResponseForbidden
+from django.shortcuts import render
 from rest_framework.decorators import api_view
 
-def get_conversation_data(conversation, user_id):
+def get_conversation_data(conversation: Conversation, user_id):
+	# Ensure given user is a participant
+	user: ConversationParticipant | None = conversation.participants.filter(user__id=user_id)[0]
+	if user is None:
+		raise Exception('The given user id does not belong to the requested conversation.')
+
 	oldest_to_look_for = datetime.datetime.now(datetime.UTC)
 	oldest_to_look_for -= datetime.timedelta(days=30)
 	messages = Message.objects \
 		.filter(conversation__id=conversation.id, date__gte=oldest_to_look_for) \
 		.order_by('-date')[:20]
 	data = {
-		'participants': [p.user.get_full_name() for p in conversation.participants.exclude(user__id=user_id).all()],
-		'messages': [
+		'participants': [
 			{
-				'message': m.text.decode(),
-				'username': m.user.get_full_name(),
-				'time': str(m.date),
-			} for m in messages[::-1]
-		],
+				'name': p.user.get_full_name(),
+				'id': p.user.id,
+	 		} for p in conversation.participants.exclude(user__id=user_id).all()],
+		'messages': [m.json_serializable() for m in messages[::-1]],
 		'conversation_id': conversation.id,
+		'key': base64.b64encode(user.encrypted_key).decode('ascii'), # Dumb library gives base64 output as bytes, so we must decode it.
```

Comment:
It's confusing looking at all the changes together like that. It makes more sense seeing changes on individual commits. Or even just looking at the plain code without any diff, I think it clearer than with the combined diffs.

Reply:
Your code should be readable, just looking at everything altogether, otherwise, you should have utilized smaller PRs, the whole idea behind a PR is that you don't need to see any single comment but it should be decently possible to read and understand the project from the changes itself